airbyte-cdk/python/airbyte_cdk/sources/file_based/config/file_based_stream_config.py

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import codecs
+from typing import Any, List, Mapping, Optional, Union
+
+from airbyte_cdk.models import ConfiguredAirbyteCatalog
+from pydantic import BaseModel, root_validator, validator
+
+PrimaryKeyType = Optional[Union[str, List[str], List[List[str]]]]
+
+
+class CsvFormat(BaseModel):
+    filetype: str
+    delimiter: str = ","
+    quote_char: str = '"'
+    escape_char: Optional[str]
+    encoding: Optional[str] = "utf8"
+    double_quote: bool
+    # Noting that the existing S3 connector had a config option newlines_in_values. This was only supported by pyarrow and not
```

Comment:
I think this can be supported using [quoting=csv.QUOTE_ALL](https://docs.python.org/3/library/csv.html#csv.QUOTE_ALL):

```
def test_csv():
    import csv

    with open('your_file.csv', newline='') as csvfile:
        reader = csv.reader(csvfile, quotechar='"', quoting=csv.QUOTE_ALL)
        # Skip the header
        next(reader)

        for row in reader:
            assert row[0] == "hello this is a string\nwith a new line"
```

with this csv file:
```
"h1","h2"
"hello this is a string
with a new line", "world"
```

Supporting different quote modes seems reasonable to me

Reply:
Ahh interesting if we were to use that option, I hadn't considered it. Although, I do worry about trying to remap the existing `newlines_in_values` to the `quoting` concept because they're not entirely 1:1. A user might not quite intend for that behavior when they choose newlines_in_values. 

wdyt if we instead just opt for a new config option that is just called `quoting` and that allows for four values: `QUOTE_ALL`, `QUOTE_MINIMAL`, `QUOTE_NONE`, `QUOTE_NONNUMERIC`? From the UI we would probably just display all, minimal, none, or nonnumeric? And maybe we still aim to deprecate the old config once we start looking into moving S3 over

Reply:
I agree we shouldn't remap it. Better to introduce the new option and deprecate `newlines_in_values`

Reply:
okay sounds good. I'll add the new quoting options